express-notes-api/index.js

```diff
@@ -0,0 +1,113 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const useJson = express.json();
+const jsonData = require('./data.json');
+let resArr = [];
+
+app.use(useJson);
+
+const pushJsonIntoArray = () => {
+  const eachNoteArr = Object.entries(jsonData.notes);
+  for (let i = 0; i < eachNoteArr.length; i++) {
+    resArr.push(eachNoteArr[i][1]);
+  }
+};
+
+app.get('/api/notes', (req, res) => {
+  resArr = [];
+  pushJsonIntoArray();
+  res.status(200).send(resArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  resArr = [];
+  pushJsonIntoArray();
+  const userRequestId = parseInt(req.params.id);
+  if (userRequestId <= 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (userRequestId < jsonData.nextId) {
+    for (let i = 0; i < resArr.length; i++) {
+      if (resArr[i].id === userRequestId) {
+        res.status(200).send(resArr[i]);
+      }
+    }
+  } else if (!userRequestId) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+
+  } else {
+    res.status(404).send({ error: `cannot find note with id ${userRequestId}` });
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  const content = req.body;
+
+  if (Object.keys(content).length === 0) {
+    res.status(400).send({ error: 'Content is a required field' });
+
```

Comment:
`{ hi: 'mil' }` will not trip this conditional

Reply:
done